### ISSUE TITLE
Benchmarking Track's WP3: What libraries do we use for benchmarks ?

### ISSUE THREAD

**Issue by jeandut** [owkin/FLamby#10]
The whole dataset API aims at being FL framework agnostic therefore we need to show examples of how to use at least 3 different FL frameworks in the repository.
In order to guide our choice of framework we need to assess:
- How easy it is to use/insert in the dataset repository ? (**all FL framework candidates need to come packaged**, be as lightweight as possible (no need for docker containers for faster benchmarking) and have pytorch support)
- Are all strategies, we want to test already implemented in that framework and easily accessible ?

Considering the small number of strategies to implement we think it might be quicker to go and implement all of them in low-level pytorch as it would not take very long.
This could be done by adding a strategy folder to the repository:
```
- strategies
     fedavg.py
     fedprox.py
     scaffold.py
     cyclic_training.py
     local.py
     ensemble.py
```
With regards to the strategy API it would take a minima a list of dataloaders, a loss, a batch-size, possibly an optimizer and some logging parameters.
If anyone has experience using other FL frameworks it would be great to benchmark different FL frameworks for each task, however a framework's result will be reported only if we have time to test it on all datasets.
